### ISSUE TITLE
[DOCS] Update API Documentation for MITRE ATT&CK subtechniques

### ISSUE THREAD

**Issue by spong** [elastic/security-docs#628]
## Description 

With the addition of MITRE Subtechniques in `7.11` (https://github.com/elastic/kibana/pull/83511), the rule creation docs were updated (https://github.com/elastic/security-docs/issues/432), but looks like we still need to update the [API Documentation](https://www.elastic.co/guide/en/security/current/rules-api-create.html#threats-object-create) still. 

@dplumlee, could you work with @jmikell821 and @nastasha-solomon on documenting this change please? 

As surfaced on [Discuss](https://discuss.elastic.co/t/detection-rules-subtechniques/269658).

**Comment by jmikell821** [comment]
````
|Name |Type |Description

|framework |String |Relevant attack framework.

|tactic |Object a|Object containing information on the attack type:

* `id` - string, required
* `name` - string, required
* `reference` - string, required

|technique |Array a|Array containing information on the attack techniques (optional): 

* `id` - string, required
* `name` - string, required
* `reference` - string, required
* `subtechnique` - Array, optional

|subtechnique |Array a|Array containing more specific, information on the attack technique:

* `id` - string, required
* `name` - string, required
* `reference` - string, required`

**Comment by jmikell821** [comment]
Merged #668.
